### PR TITLE
Timeout improvements in Upgrade

### PR DESCRIPTION
**Pull request by jyejare**
Timeout improvements in Upgrade Automation!

The number trys/attempts for timeout has been replaced with time in minutes.

### CODE REVIEW

File: automation_tools/__init__.py [hunk 1]
```diff
@@ -1644,15 +1644,16 @@ def setenforce(mode):
...
 
     :param host: A string. The IP or hostname of host.
-    :param attempts: An integer.
-        The maximum number of attempts to ping the host.
+    :param timeout: An integer.
+        The timeout in minutes to ping the host.
 
     """
-    for attempt in range(attempts):
+    timeup = time.time()+(int(timeout)*60)
```
Comment (Ichimonji10) [inline automation_tools/__init__.py]: Python's operator precedence is the same as in algebra: multiplication and division before addition and subtraction.

```python
>>> 1 + 2 * 10
21
```

Nitpick: Add some whitespace around each operator and drop unnecessary parentheses:

```python
timeup = time.time() + int(timeout) * 60
```

File: automation_tools/__init__.py [hunk 2]
```diff
@@ -1763,22 +1774,25 @@ def create_openstack_instance(instance_name, image_name, flavor_name, ssh_key):
         network=network.id
     )
     # Assigning floating ip to instance
+    timeup = time.time()+(int(timeout)*60)
```
Comment (Ichimonji10) [inline automation_tools/__init__.py]: Same nitpick.

File: automation_tools/__init__.py [hunk 3]
```diff
@@ -1763,22 +1774,25 @@ def create_openstack_instance(instance_name, image_name, flavor_name, ssh_key):
...
     while True:
         try:
+            if time.time() > timeup:
+                print('The timeout for assigning the floating IP has reached!')
+                sys.exit(1)
             instance.add_floating_ip(floating_ip)
+            print('SUCCESS!! The flaoting IP has been assigned to instance!')
```
Comment (Ichimonji10) [inline automation_tools/__init__.py]: "flaoting"

Comment (Ichimonji10) [inline automation_tools/__init__.py]: Would it be helpful to note information about `floating_ip` or `instance` in these messages?

File: automation_tools/__init__.py [hunk 4]
```diff
@@ -1763,22 +1774,25 @@ def create_openstack_instance(instance_name, image_name, flavor_name, ssh_key):
         network=network.id
     )
     # Assigning floating ip to instance
+    timeup = time.time()+(int(timeout)*60)
     while True:
         try:
+            if time.time() > timeup:
+                print('The timeout for assigning the floating IP has reached!')
+                sys.exit(1)
```
Comment (Ichimonji10) [inline automation_tools/__init__.py]: Does this need to be in the `try` block? If reasonably possible, only code that might raise a `novaclient.exceptions.BadRequest` should be in the `try` block.

**Comment by Ichimonji10** [comment]
ACK pending comments.

File: automation_tools/__init__.py (hunk 1 above)
Comment (jyejare) [inline automation_tools/__init__.py]: +1. Done

File: automation_tools/__init__.py (hunk 2 above)
Comment (jyejare) [inline automation_tools/__init__.py]: Done.

File: automation_tools/__init__.py (hunk 3 above)
Comment (jyejare) [inline automation_tools/__init__.py]: Noted only Floating IP. The instance name is not required note as we are not using it anywhere to test.

File: automation_tools/__init__.py (hunk 4 above)
Comment (jyejare) [inline automation_tools/__init__.py]: +1. Done.

**Comment by elyezer** [comment]
ACK

**Pull request merged**